site/src/components/Workspace/Workspace.tsx

```diff
@@ -131,6 +132,43 @@ export const Workspace: FC<React.PropsWithChildren<WorkspaceProps>> = ({
   if (template !== undefined) {
     transitionStats = ActiveTransition(template, workspace)
   }
+
+  const [showAlertPendingInQueue, setShowAlertPendingInQueue] = useState(false)
+  const now = dayjs()
+  useEffect(() => {
+    if (
+      workspace.latest_build.status !== "pending" ||
+      workspace.latest_build.job.queue_size === 0
+    ) {
+      if (!showAlertPendingInQueue) {
+        return
+      }
+
+      // hideTimer
+      setTimeout(() => {
+        setShowAlertPendingInQueue(false)
+      }, 250)
+      return
```

Comment:
I wonder if we need to return a `clearTimeout` here too? Otherwise navigating away from this view could trigger the timer after-the-fact (is that a problem?). Then again, if the function is triggered every time `useEffect` has new values, we need to make sure we don't repeatedly set/clear the timeout when e.g. new workspace data comes in (let's say there are 10 updates <250ms apart each, then this would ultimately take ~2.5s to trigger).

Reply:
I played a bit, and didn't observe any unexpected side effects of having `clearTimeout`, so added it 👍